pkg/machine/qemu/machine.go

```diff
@@ -927,7 +927,11 @@ func (v *MachineVM) collectFilesToDestroy(opts machine.RemoveOptions) ([]string,
 	files := []string{}
 	// Collect all the files that need to be destroyed
 	if !opts.SaveKeys {
-		files = append(files, v.IdentityPath, v.IdentityPath+".pub")
+		confDir := filepath.Dir(v.ConfigPath.Path)
+
+		if machine.HostHasMultipleMachines(new(MachineVM), confDir) {
```

Comment:
Should this be 
```suggestion
		if !machine.HostHasMultipleMachines(new(MachineVM), confDir) {
```

since we want to keep the keys, ie not append to the deleted files if there are multiple machines? Ditto for other instances in the other providers